xesmf/smm.py

```diff
@@ -115,3 +116,19 @@ def apply_weights(weights, indata, shape_in, shape_out):
     outdata = outdata_flat.reshape(
         [*extra_shape, shape_out[0], shape_out[1]])
     return outdata
+
+
+def add_nans_to_weights(weights):
+    """ Add NaN in empty rows of the regridding weights sparse matrix,
+        useful for turning masked values from zero to NaN. Original code
+        from @trondkr, adapted by @raphaeldussin to use lil"""
```

Comment:
```suggestion
    """Add NaN in empty rows of the regridding weights sparse matrix.
    
    By default, empty rows in the weights sparse matrix are interpreted as zeroes. This can become problematic 
    when the field being interpreted has legitimate null values. This function inserts NaN values in each row to 
    make sure empty weights are propagated as NaNs instead of zeros. 
    
    Parameters
    ----------
    weights : ...
"""
# Taken from @trondkr and adapted by @raphaeldussin to use `lil`.
```